natlas-server/app/elastic/interface.py

```diff
@@ -5,12 +5,17 @@
 
 class ElasticInterface:
     client = None
+    default_indices = {"latest": "nmap", "history": "nmap_history"}
 
-    def __init__(self, elasticUrl):
-        self.client = ElasticClient(elasticUrl)
+    def __init__(self, elasticUrl, indices=default_indices):
+        self.indices = indices
+        self.indices_str = ",".join(indices.values())
```

Comment:
Another design that I'm currently weighing is the name provider pattern.

Example:
```python
class ElasticIndexNames:
    def name(self, alias):
       return blah
    def all_indices(self):
       return []
```

With this strategy, we can pull this logic out into a separate micro class and can then enforce checks like: Is this a valid index alias or could even do something like a dynamic time based alias selection which is a common ES pattern.

Thoughts?

Reply:
I took this idea but I expanded it a little further to make one class that handles all the index related activity. Initializing them, deleting them (only used in tests right now), and return indexes related to the current context we're in. I think this breaks the provider pattern maybe? But it does create a more clear distinction between the index operations and the dumber "client" class. This way index names are completely centralized into one place, the ElasticIndices class.

I also took this opportunity to add type hinting to the elastic classes where ever I could